tests/tests/lib_tests.rs

```diff
@@ -14,532 +14,240 @@ use wasmtime::Val;
 use clar2wasm_tests::datastore::{BurnDatastore, Datastore, StacksConstants};
 use clar2wasm_tests::{ClarityWasmResult, WasmtimeHelper};
 
-#[test]
-fn add() {
-    let contract_id = QualifiedContractIdentifier::new(
-        StandardPrincipalData::transient(),
-        ContractName::from("add"),
-    );
-    let mut datastore = Datastore::new();
-    let constants = StacksConstants::default();
-    let burn_datastore = BurnDatastore::new(constants);
-    let mut conn = ClarityDatabase::new(&mut datastore, &burn_datastore, &burn_datastore);
-    conn.begin();
-    conn.set_clarity_epoch_version(StacksEpochId::Epoch24);
-    conn.commit();
-    let cost_tracker = LimitedCostTracker::new_free();
-    let mut global_context = GlobalContext::new(
-        false,
-        CHAIN_ID_TESTNET,
-        conn,
-        cost_tracker,
-        StacksEpochId::Epoch24,
-    );
-    let mut contract_context = ContractContext::new(contract_id.clone(), ClarityVersion::Clarity2);
-
-    global_context.begin();
-    {
-        let mut helper =
-            WasmtimeHelper::new_from_file(contract_id, &mut global_context, &mut contract_context);
-
-        if let ClarityWasmResult::Response {
-            indicator,
-            ok_value,
-            err_value,
-        } = helper.call_public_function("simple", &[])
-        {
-            assert_eq!(indicator, 1);
-            assert!(ok_value.is_some());
-            assert!(err_value.is_none());
-            if let ClarityWasmResult::Int { high, low } = *ok_value.unwrap() {
-                assert_eq!(high, 0);
-                assert_eq!(low, 3);
+type Res = Option<Box<ClarityWasmResult>>;
+
+/// This macro provides a convenient way to test functions inside contracts.
+/// In order, it takes as parameters:
+/// - the name of the test to create,
+/// - the name of the contract containing the function,
+/// - the name of the function to test,
+/// - an optional list of parameters,
+/// - a closure with type `|indicator: i32, ok_value: Res, err_value: Res|`, and
+///   that contains all the assertions we want to test.
+macro_rules! test_contract {
+    ($func: ident, $contract_name: literal, $contract_func: literal, $params: expr, $test: expr) => {
+        #[test]
+        fn $func() {
+            let contract_id = QualifiedContractIdentifier::new(
+                StandardPrincipalData::transient(),
+                ContractName::from($contract_name),
+            );
+            let mut datastore = Datastore::new();
+            let constants = StacksConstants {
```

Comment:
This can use the `default` method from PR #36

Reply:
Oh no, I guess I lost it during my rebase... I will change that.